xdsl/parser/attribute_parser.py

```diff
@@ -651,6 +652,22 @@ def _parse_strided_layout_attr(self, name: Span) -> Attribute:
         self._parse_token(Token.Kind.GREATER, "Expected '>' in end of stride attribute")
         return StridedLayoutAttr(strides, None if offset == "?" else offset)
 
+    def parse_optional_unit_attr(self) -> Attribute | None:
+        """
+        Parse a value of `unit` type.
+        unit-attribute ::= `unit`
+        """
+        if self._current_token.kind != Token.Kind.BARE_IDENT:
+            return None
+        name = self._current_token.span.text
```

Comment:
supernit: Not sure this extra variable assignment is needed here?

Reply:
definitely not, although it doesn't really hurt either